[JSC] Wasm GC: Simplify RTT::isStrictSubRTT C++ and JIT code

dhecht opened the pull request:
#### d3343c8e5a974fabc1af6cb1a75741997ff71cb0
<pre>
[JSC] Wasm GC: Simplify RTT::isStrictSubRTT C++ and JIT code
<a href="https://bugs.webkit.org/show_bug.cgi?id=294850">https://bugs.webkit.org/show_bug.cgi?id=294850</a>
<a href="https://rdar.apple.com/154099633">rdar://154099633</a>

Reviewed by Keith Miller.

The special case for parent.displaySize() == 0 is unnecessary because
when parent.displaySize() is 0, then:
1. displaySize() &lt;= parent.displaySize() is always false (since
   displaySize() &gt; 0)
2. displaySize() - parent.displaySize() - 1 is equivalent to displaySize() - 1)

Then, we can further simplify since displaySize() check becomes
redundant with displaySize() &lt;= parent.displaySize().

So simplify this code and remove the unnecessary branches.

Canonical link: <a href="https://commits.webkit.org/296535@main">https://commits.webkit.org/296535@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0ebcde7d5dcd4da51074b3188170610490e2b8c2

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/108838 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/28498 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/18923 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/114046 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/59195 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/29182 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/37064 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/82696 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/59195 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/111786 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/23188 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/98030 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/63135 "Passed tests") | 
| | [⏳ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/iOS-18-Simulator-WPT-WK2-Tests-EWS "Waiting to run tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/16167 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/58749 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/101378 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/92560 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/16215 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/117167 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/107397 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/35888 "Built successfully") | [⏳ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/macOS-Sequoia-Debug-WK2-Tests-EWS "Waiting to run tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/91714 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/36261 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/94296 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/91522 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/23297 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/36420 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/14176 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/31758 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/35787 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/41318 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/131673 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/35488 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/35711 "Passed tests") | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/129/builds/38833 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/37174 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/b0a94e13411985cd30a53ee000ef81186d22f47a. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/47068

dhecht commented on Source/JavaScriptCore/wasm/WasmTypeDefinition.cpp:
```diff
@@ -584,17 +584,11 @@ RefPtr<RTT> RTT::tryCreateRTT(RTTKind kind, DisplayCount displaySize)
-
-    return false;
+    if (!displaySize())
+        return false;
```
Done

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/177ce122d443862d84c600bff9f93aa27704353b. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/47068

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/ac8f32f6d3d1715c18f1941516b8913183b4886f. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/47068

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/0ebcde7d5dcd4da51074b3188170610490e2b8c2. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/47068

dhecht committed:
[JSC] Wasm GC: Simplify RTT::isStrictSubRTT C++ and JIT code
https://bugs.webkit.org/show_bug.cgi?id=294850
rdar://154099633

Reviewed by Keith Miller.

The special case for parent.displaySize() == 0 is unnecessary because
when parent.displaySize() is 0, then:
1. displaySize() <= parent.displaySize() is always false (since
   displaySize() > 0)
2. displaySize() - parent.displaySize() - 1 is equivalent to displaySize() - 1)

Then, we can further simplify since displaySize() check becomes
redundant with displaySize() <= parent.displaySize().

So simplify this code and remove the unnecessary branches.

Canonical link: https://commits.webkit.org/296535@main
